src/ilasm/assembler.cpp

```diff
@@ -126,8 +126,16 @@ mdToken Assembler::ResolveClassRef(mdToken tkResScope, __in __nullterminated con
             }
             return tkRet;
         }
-        else  // needs to be resolved
-            if(!m_fIsMscorlib) tkResScope = GetBaseAsmRef();
+        else
+        { // needs to be resolved
+            if (TypeFromToken(tkResScope) == mdtAssemblyRef && RidFromToken(tkResScope) > 0)
+            {
+                // considering the aliased case '[' dottedName ']' slashedName
+                AsmManAssembly* assembly = m_pManifest->m_AsmRefLst.PEEK(RidFromToken(tkResScope) - 1);
```

Comment:
Regarding #2, can you attempt to construct a case with multiple levels of aliases and see what you get?  If the syntax doesn't allow you to create an alias of SomethingElse (from original repro code), then I think you may be ok and this change looks good to me.